lora_python_log_serial_CURRENT.py

```diff
@@ -1,74 +1,106 @@
+#!/usr/bin/python3
 # author: Brandon
 # V1.0-initial,  assumes the data being transmitted is text-based. Binary data requires modification
 # V1.1-modify to handle binary data & write to file more frequently
-# V1.3-change delimeter to ' to make parsing easier
-# V1.4-added log_version variable
-
-#@TODO:
-#add ability to reset the radio via GPIO
-
+# V1.2- Collin's cleanup.
+import json
+import logging
+import re
 import serial
-from datetime import datetime
 import time
-
-# general variables
-debug_on = True
-log_version = 1.4 #@TODO-get this to save log version into the file
+from pathlib import Path
 
 # Serial port settings
 serial_port = "/dev/ttyS0"  # Replace with the appropriate serial port path
-#@TODO-fix the problem of message corruption when throttling is throttled=0x80008 or throttled=0xe000e. 300 and 600 give \x00\x00\ over and over. setting to 1200 gives good data when the system isn't throttling. set back to 9600
-baud_rate = 9600  # Adjust to match the transmitting device's baud rate. 
+baud_rate = 9600  # Adjust to match the transmitting device's baud rate
 timeout_seconds = 0.05  # time it will wait for a newline. 0.05 secs is more than enough time for ~30 characters of serial data to transmit at 9600 baud.
 
 # File settings
-log_file = "/opt/lora_logger/serial_log_python_lora.txt"
-# Time interval for flushing aka forcing write of log to disk
-flush_interval = 30 #in seconds
-# Variables to track time
-last_flush_time = time.time()
+base_dir = Path("/opt/lora_logger")
+log_file = base_dir / Path("lora_python_serial_log_V1.2.txt")
 
-print("lora_python_log_serialV1.1.py-beginning to log serial data. Anything received will go in file.")
-print("log version:")
-print(log_version)
+# DEBUG -> INFO -> WARNING -> ERROR -> CRITICAL
+# Logging logs every level above the current one.
+logging.basicConfig(
+    filename=log_file,
+    format="%(asctime)s %(message)s",
+    filemode="a",
+    encoding="utf-8",
+    level=logging.DEBUG,
+)
 
 
-def get_timestamp():
-    return datetime.now().strftime("'%Y'%m'%d'%H'%M'%S'")
+def parse_line(l):
+    splits = [s for s in re.split("['\s]", l) if s]
+    message_info = re.split(",", splits[-1])
+    #logging.info(splits)
+    #logging.info(message_info)
```

Comment:
Don't comment code out on git. Old code will be in the git history so there's no need to commit commented code.